packages/core/src/core.ts

```diff
@@ -11,3 +11,71 @@ export * from "./util/pascalCaseToKebabCase.js"
 export * from "./util/pluralize.js"
 export * from "./util/singularize.js"
 export * from "./util/uuidv4.js"
+
+// required to explicitly export types for inference with esm modules!!
```

Comment:
please link to [this issue](https://github.com/microsoft/TypeScript/issues/47663) from this comment and mention that in this specific case it is a workaround for TS failing to resolves these types correctly through several nested `export * from blah` references.

Reply:
now that this is in `dataTypes/index.ts` do we still need this comment? previously it just _wasn't_ re-exported, though that was fine before. these types aren't expected to be used by the user, they are just required for TS when calling the function since it can't figure out the return type if we aren't re-exporting

Reply:
nope, this finally solves it once and for all. 🎉